### ISSUE TITLE
Love 0.10.0 for Windows: --console doesn't seem to work

### ISSUE THREAD

**Issue by footballhead** [footballhead/love-launcher#8]
This thread is to investigate why `--console` no longer seems to work in LOVE 0.10.0 with the purpose of improving this software as well as possibly filing an upstream bug report.

Thanks to @endlesstravel for the initial report.

Environment
- Windows 10, x64

So far:
- `--console` works for 0.9.1 (timestamp around 2014-04-01)
- it **doesn't** work for 0.9.2 (timestamp around 2015-02-26)
- according to the changelog (https://love2d.org/wiki/0.9.2) console behaviour slightly changed between 0.9.1 and 0.9.2... this change probably caused the breakage we see currently

Next steps:
- dive into the love code to see what changed between those two versions (between 2014-04-01 and 2015-02-26)

**Comment by pablomayobre** [comment]
This issue is because with `BufferedProcess` you are actually grabbing `stdout` and `stderr` of LÖVE, so when you ask LÖVE for console output LÖVE uses the `BufferedProcess` instead of spawning a new one, this is the behaviour that changed in 0.9.2, before LÖVE ignored the console it was spawned from and created a new one, this behaviour will change again with LÖVE 0.10.2 and 0.11.0 with the addition of `lovec.exe` which has it's own terminal or something like that, not sure how it will work though.

LÖVE dumps `stdout` to the `BufferedProcess` on closing unless you have `io.stdout:setvbuf('no')` somewhere in you Lua code. This is how Lua works by default so it's expected.

An easy fix for the issue, that is to spawn a new console, would be to use a non buffered process, which you may be able to do using Node.js `ChildProcess`, syntax should be similar, but I haven't investigated nor tested this.

**Comment by footballhead** [comment]
Commit 52c17805939dcf02d779231da6d2d2c223048b9d fixes this for Windows. On Linux, however, --console doesn't launch an extra window, so we relied on capturing stdout to print to the developer console. `ChildProcess` can still be used, but (as far as I understand) it can't be detached process, and an event handler will need to be registered for stdout data.

**Comment by footballhead** [comment]
something like:

```coffeescript
process.stdout.on("data", (data) -> console.log(data.toString()) )
```

**Comment by footballhead** [comment]
This should be fixed with version 0.7.0. If not we can reopen and reinvestigate
